pkg/services/auth/authimpl/auth_token.go

```diff
@@ -574,6 +575,59 @@ func (s *UserAuthTokenService) GetUserRevokedTokens(ctx context.Context, userId
 	return result, err
 }
 
+func (s *UserAuthTokenService) GetUserRevokedTokensInLastHours(ctx context.Context, userId int64, duration time.Duration) ([]*auth.UserToken, error) {
+	hours := int(duration.Hours())
+	if hours < 1 {
+		s.log.Warn("GetUserRevokedTokensInLastHours called with duration less than 1 hour", "duration", duration)
+		hours = 1
+	}
+	result := []*auth.UserToken{}
+	err := s.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		var tokens []*userAuthToken
+		// get the database, mysql, postgres or sqlite
+		dialect := s.sqlStore.GetDialect()
+		var query string
+
+		if dialect.DriverName() == migrator.Postgres {
+			// postgres
+			query = fmt.Sprintf("user_id = ? AND revoked_at > 0 AND revoked_at <= EXTRACT(EPOCH FROM (NOW() - INTERVAL '%d hours'))::INTEGER", hours)
+		} else if dialect.DriverName() == migrator.MySQL {
+			// mysql
+			// example: select UNIX_TIMESTAMP(NOW() - INTERVAL 20 HOUR) # 1694362241
+			query = fmt.Sprintf("user_id = ? AND revoked_at > 0 AND revoked_at <= UNIX_TIMESTAMP(NOW() - INTERVAL %d HOUR", hours)
+		} else if dialect.DriverName() == migrator.SQLite {
+			// SQLite does not have native support for date arithmetic like other SQL dialects,
+			// strftime gets the Unix timestamp
+			query = fmt.Sprintf("user_id = ? AND revoked_at > 0 AND revoked_at <= (strftime('%%s','now') - %d*3600)", hours)
+		} else if dialect.DriverName() == migrator.MSSQL {
```

Comment:
We don't support mssql so we can remove this case

Reply:
we have it as a migrator, should we remove that in a separate pr?